clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp

```diff
@@ -253,6 +253,17 @@ class TrivialFunctionAnalysisVisitor
     return true;
   }
 
+  template <typename StmtType, typename CheckFunction>
+  bool WithCachedResult(const StmtType *S, CheckFunction Function) {
```

Comment:
```suggestion
  template <typename CheckFunction>
  bool WithCachedResult(const Stmt *S, CheckFunction Function) {
```
A bit simpler since we don't really care about the sub-class.

Reply:
Good point. Changed.